components/filecoin/rs/src/signature.rs

```diff
@@ -0,0 +1,187 @@
+use crate::message::MessageAPI;
+use blake2b_simd::Params;
+use bls_signatures::Serialize;
+use core::{array::TryFromSliceError, num::ParseIntError};
+use fvm_ipld_encoding::Cbor;
+use fvm_shared::address::Protocol;
+use fvm_shared::message::Message as UnsignedMessage;
+use libsecp256k1::util::{SECRET_KEY_SIZE, SIGNATURE_SIZE};
+use libsecp256k1::{sign, Message};
+use std::convert::TryInto;
+use thiserror::Error;
+
+pub const SIGNATURE_RECOVERY_SIZE: usize = SIGNATURE_SIZE + 1;
+pub const BLS_SIGNATURE_SIZE: usize = 96;
+
+pub struct PrivateKey(pub [u8; SECRET_KEY_SIZE]);
+pub struct SignatureBLS(pub [u8; BLS_SIGNATURE_SIZE]);
+pub struct SignatureSECP256K1(pub [u8; SIGNATURE_RECOVERY_SIZE]);
+
+pub enum Signature {
+    SignatureSECP256K1(SignatureSECP256K1),
+    SignatureBLS(SignatureBLS),
+}
+
+impl Signature {
+    pub fn as_bytes(&self) -> &[u8] {
+        match self {
+            Signature::SignatureSECP256K1(sig_secp256k1) => sig_secp256k1.as_bytes(),
+            Signature::SignatureBLS(sig_bls) => sig_bls.as_bytes(),
+        }
+    }
+}
+
+impl SignatureBLS {
+    pub fn as_bytes(&self) -> &[u8] {
+        &self.0
+    }
+}
+
+impl SignatureSECP256K1 {
+    pub fn as_bytes(&self) -> &[u8] {
+        &self.0
+    }
+}
+
+/// Filecoin Signer Error
+#[derive(Error, Debug)]
+pub enum SignerError {
+    /// CBOR error
+    #[error("CBOR error: '{0}'")]
+    CBOR(#[from] serde_cbor::Error),
+    /// Secp256k1 error
+    #[error("secp256k1 error")]
+    Secp256k1(#[from] libsecp256k1::Error),
+    // Key decoding error
+    #[error("key decoding error (only hex or base64 is accepted)")]
+    KeyDecoding(),
+    /// Hex Error
+    #[error("Hex decoding error | {0}")]
+    HexDecode(#[from] hex::FromHexError),
```

Comment:
I don't see where this error is being thrown. Is this happening within a library that we call? Otherwise I think we can get rid of this dependency

Reply:
If this isn't needed, the `hex` crate dependency should be removed from `Cargo.toml` as well.

Reply:
removed